### ISSUE TITLE
An option to disable boot timeout or set timeout time

### ISSUE THREAD

**Issue by lkocman** [agama-project/agama#1594]
Hello team

I was just looking where to disable bootloader timeout (quite important for openQA for reboot after install) and well aside from that not being an option, I noticed that it's ver well hidden and I really had to be creative to find it :-)

Aside from this particular openQA  case I think generally people want some timeout for power on lan, reboot after outage etc, automatic updates etc ... I could alternatively set a long timeout, in case that there is some value that can be set.

![image](https://github.com/user-attachments/assets/6ce08e0b-41f3-413f-8427-145212f7115b)

![image](https://github.com/user-attachments/assets/10d010bf-4efd-49eb-b56d-362122f6f2b6)

![image](https://github.com/user-attachments/assets/f032f0e1-0a94-41e0-8f6d-a066828425ae)

**Comment by dgdavid** [comment]
> I was just looking where to disable bootloader timeout (quite important for openQA for reboot after install) and well aside from that not being an option, I noticed that it's ver well hidden and I really had to be creative to find it :-)

Hi @lkocman

AFAIK, there is not such an option (yet). About the other option you mention, 

* it's well hidden on purpose, since we understood it is a very special case
* be aware that it is not the option you're looking for

Let me ping @ancorgs to help us here.

**Comment by dgdavid** [comment]
Thanks @ancorgs!

I'm a bit curious to know why such an option was useful for @lkocman as a _workaround_ for the bootloader timeout.

**Comment by lkocman** [comment]
@dgdavid This it's not workaround, setting timeout during installation to 0 is what openqa generally expects.
 Just to let you know, the workaround is to hit spacebar every 3 seconds or until grub appears. Otherwise, in many cases we were simply not fast enough to catch the screen and cancel timeout. Prefered would be to add a step to disable timeout just like with traditional installer,.

**Comment by dgdavid** [comment]
> @dgdavid This it's not workaround, 

So, I miss-understood your first comment :sweat_smile: 

> Just to let you know, the workaround is to hit spacebar every 3 seconds or until grub appears. Otherwise, in many cases we were simply not fast enough to catch the screen and cancel timeout. 

Thanks for the explanation.


> Prefered would be to add a step to disable timeout just like with traditional installer,.

I see. But as @ancorgs said it needs some discussion first.

**Comment by lkocman** [comment]
I agree, we're not blocked any more, but in general I think we should have this feature.

**Comment by jreidinger** [comment]
@lkocman disabling timeout is not implemented with https://github.com/agama-project/agama/pull/1840 ( it is enough for openqa to do it on CLI without webui ). We just get follow up that in some cases they need to set high value when it is real hardware with unreliable IPMI, so there is potential followup.
